python/cuvs/cuvs/neighbors/ivf_flat/ivf_flat.pyx

```diff
@@ -0,0 +1,354 @@
+#
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# cython: language_level=3
+
+import numpy as np
+
+cimport cuvs.common.cydlpack
+
+from cuvs.common.resources import auto_sync_resources
+
+from cython.operator cimport dereference as deref
+from libcpp cimport bool, cast
+
+from cuvs.common cimport cydlpack
+from cuvs.distance_type cimport DistanceType
+
+from pylibraft.common import auto_convert_output, cai_wrapper, device_ndarray
+from pylibraft.common.cai_wrapper import wrap_array
+from pylibraft.common.interruptible import cuda_interruptible
+from pylibraft.distance.pairwise_distance import DISTANCE_TYPES
+from pylibraft.neighbors.common import _check_input_array
+
+from libc.stdint cimport (
+    int8_t,
+    int64_t,
+    uint8_t,
+    uint32_t,
+    uint64_t,
+    uintptr_t,
+)
+
+from cuvs.common.exceptions import check_cuvs
+
+
+cdef class IndexParams:
+    """
+    Parameters to build index for IvfFlat nearest neighbor search
+
+    Parameters
+    ---------
+    n_lists : int, default = 1024
+        The number of clusters used in the coarse quantizer.
+    metric : string denoting the metric type, default="sqeuclidean"
+        Valid values for metric: ["sqeuclidean", "inner_product",
+        "euclidean"], where
+            - sqeuclidean is the euclidean distance without the square root
+              operation, i.e.: distance(a,b) = \\sum_i (a_i - b_i)^2,
+            - euclidean is the euclidean distance
+            - inner product distance is defined as
+              distance(a, b) = \\sum_i a_i * b_i.
+    kmeans_n_iters : int, default = 20
+        The number of iterations searching for kmeans centers during index
+        building.
+    kmeans_trainset_fraction : int, default = 0.5
```

Comment:
I wonder if adding a suggestion of when to change this would be useful for users

Reply:
I think it could be very helpful!  Though - I'm not sure of when to change this myself, and have just copied this comment from the c++ api. Do you have any thoughts here?

Reply:
@cjnolet that sounds good! I've added in the last commit https://github.com/rapidsai/cuvs/pull/82/commits/3c9efd3832a4edc312e390dd2cb33267f802c487 (which also adds this into the docs site)